File: Kotlin/TwoSum.kt
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val hashMap = mutableMapOf<Int, Int>()
        val arr = arrayListOf<Int>()

        for (i in nums.indices) {
            hashMap[i] = nums[i]
        }

        for ((key, value) in hashMap) {
            val complement = target - value

            for ((k, v) in hashMap) {
                if (v === complement && k != key) {
                    arr.add(key)
                    arr.add(k)
                    return arr.toIntArray()
                }
            }
        }
        return arr.toIntArray()
    }
}

//8.26
class Solution {
    fun twoSum(nums: IntArray, target: Int): IntArray {
        val arr = mutableListOf<Int>()
        val map = mutableMapOf<Int, Int>()

        for(i in nums.indices) {
            map.set(nums[i], i)
        }

        for(i in nums.indices) {
            val complement = target - nums[i]

            if(map.containsKey(complement) && i != map.get(complement)) {
                    arr.add(i)
                    arr.add(map.getValue(complement))
                    return arr.toIntArray()
            }
                    
                } 
                 return arr.toIntArray()
            }
        }
      

File: JavaScript/TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
const twoSum = (nums, target) => {
    const hashMap = new Map()
    let arr = []

    for (let i = 0; i < nums.length; i++) {
        hashMap.set(i, nums[i])
    }

    for (const [key, value] of hashMap) {
        const complement = target - value

        for (const [k, v] of hashMap) {
            if (v === complement && k != key) {
                arr.push(key)
                arr.push(k)
                return arr
            }
        }
    }
}

// 8.26
const twoSum = (nums, target) => {
    //create new array
    const arr = []
    //create new map of nums and their indices
    const map = new Map()
    //loop through nums
    for (let i = 0; i < nums.length; i++) {
        map.set(nums[i], i)
    }
    //for each value
    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i]

        if (map.has(complement) && i != map.get(complement)) {
            arr.push(i)
            arr.push(map.get(complement))
            return arr
        }
    }
}